Doc/library/shutil.rst

```diff
@@ -209,14 +209,20 @@ Directory and files operations
 
 
 .. function:: copytree(src, dst, symlinks=False, ignore=None, \
-              copy_function=copy2, ignore_dangling_symlinks=False)
+              copy_function=copy2, ignore_dangling_symlinks=False, \
+              dirs_exist_ok=False)
 
    Recursively copy an entire directory tree rooted at *src*, returning the
-   destination directory.  The destination
-   directory, named by *dst*, must not already exist; it will be created as
-   well as missing parent directories.  Permissions and times of directories
-   are copied with :func:`copystat`, individual files are copied using
-   :func:`shutil.copy2`.
+   destination directory. If *dirs_exist_ok* is false (the default), then the
+   destination directory, named by *dst*, must not already exist; it will be
+   created as well as missing parent directories.  Permissions and times of
+   directories are copied with :func:`copystat`, individual files are copied
+   using :func:`shutil.copy2`.
+
+   If *dirs_exist_ok* is true, then the destination directory may already exist
+   (as may any subdirectories of the destination directory with the same names
+   as subdirectories of the source directory, since :func:`copytree` is called
+   recursively with the same options).
```

Comment:
Uhm... this looks a bit too verbose to me. Also, the explanation of "dirs_exist_ok" is split in two separate paragraphs which are interrupted by the part re. the permissions. I would rephrase it as something along these lines:

<<Recursively copy an entire directory tree rooted at *src* to a directory named dst* and return the destination directory. *dirs_exist_ok* dictates whether to raise an exception in case *dst* as well as any missing parent directory already exists.>>

Reply:
Updated in the latest revision.